plugins/bcli.c

```diff
@@ -833,14 +790,10 @@ static void parse_getnetworkinfo_result(struct plugin *p, const char *buf)
 			      gather_args(bitcoind, "getnetworkinfo", NULL), buf);
 
 	/* Check that we have a fully-featured `estimatesmartfee`. */
-	versiontok = json_get_member(buf, result, "version");
-	if (!versiontok)
-		plugin_err(p, "No 'version' in '%s' ? Got '%s'. Can not"
-			      " continue without proceeding to sanity checks.",
-			      gather_args(bitcoind, "getnetworkinfo", NULL), buf);
-
-	if (!json_to_u32(buf, versiontok, &bitcoind->version))
-		plugin_err(p, "Invalid 'version' in '%s' ? Got '%s'. Can not"
+	if (!json_scan(buf, result, "{version:%,localrelay:%}",
+		       JSON_SCAN(json_to_u32, &bitcoind->version),
+		       JSON_SCAN(json_to_bool, &tx_relay)))
+		plugin_err(p, "No 'version' or localrelay in '%s' ? Got '%s'. Can not"
```

Comment:
This error message can appear when `localrelay` is not present, which is confusing.